### ISSUE TITLE
Looking up instance properties in the declarative model

### ISSUE THREAD

**Issue by joeseibel** [osate/osate2-core#875]
As implemented in `InstanceObjectImpl.getPropertyValueInternal(Property, PropertyAcc, boolean)`, property lookup on an `InstanceObject` follows these steps:

1. Lookup locally on the `InstanceObject`
2. Lookup on the corresponding declarative object
3. If the property is inherit, lookup on the container

Is this correct? Does it even make sense to lookup a property on the declarative object when it isn't already on the instance object? Also, the instantiator already propagates inherit properties to child objects, so there should be no need for the lookup to consider inherit. This needs to be seriously considered to determine what should be done. If we decide to not go to the declarative, then we can probably get rid of the boolean flag `fromInstanceSlaveCall`.

This came up when discussing #854.

**Comment by philip-alldredge** [comment]
FYI. The graphical editor currently uses the fromInstanceSlaveCall flag for its property lookup process.  It sets the flag to true so that it can ignore contained property associations. The graphical editor processes the contained property associations itself since diagrams forms a pseudo instance model.

**Comment by reteprelief** [comment]
The code handles the situation where not all properties get recorded in the instance model. You could specify a list of properties to be copied into the instance model. Be default this list is the list of all used properties. The idea about the list was to be able to generate an instance model populated only with those properties that are relevant  for a specific analysis and still operate on the instance model when some properties are not cached in the instance model.

**Comment by lwrage** [comment]
I don't think that it makes any sense to use properties from the declarative model in the instance model as that completely ignores contained property associations. **Any** result obtained by processing these values would potentially be wrong.
